docs/source/developer/adr/0012-claims-role-scope-rework.md

```diff
@@ -0,0 +1,72 @@
+ - When building the API spec inside flowAPI
+ - When displaying the spec on the Flowauth frontend for servers (leading to timeouts and crashes on some lower-capability machines)
+ - When passing large or specific scopes around as JWTs
+
+**Decision**
+
+
+The core of the change that has been implemented is to change the definition of what a user can access. Instead of a combination of queries, users are now assigned `roles`, which are composed of `scopes`. A scope can be one of two types;
+ - *Simple scopes*
+
+   These control access to a server-wide capability and are all-or-nothing. The currently implemented simple scopes are:
+   - `run`; permits a user that olds this role to run queries using the `run` flowAPI endpoint
+   - `get_result`; permits a user to get the results of a query using the `get` flowAPI endpoint
+   - `get_available_dates`; permits a user to get the list of dates available to run queries against on a server.
+   
+   These scopes do *not* imply that a role provided the capabilites to run a query by themselves - they must be provided in combination with one or more complex scopes.
+
+ - *Complex scopes*
+  
+   These control access to a combination of a geographic component, a top-level query and a descendent query.
+   - Top-level queries are the methods that are avilable to the users of Flowmachine directly.
+   - Each top-level query may require a set of sub-queries to run, wich in turn may require sub-queries of their own; these are the descendent queries.
+   - Finally, the geographic component is the spatial presentation that is available to the role.
+   
+   Complex scopes are of the form `geographic_component:top_level_query:sub_query` - this ordering has been built on assumed order of importance to users. For example it will be a more common use case that an administrator will want to create a role that restrics access only to admin levels 0 or 1 than to the `most_frequest_location` sub-query of the `spatial_aggregate` TL query.
+
+As a consequence of this, the new Flowauth db schema is shown below:
+
+```mermaid
+erDiagram
+    Servers ||--|{ Roles : contains
+    Roles }|--|{ Scopes : "provides permissions for"
+    Users }|--|{ Roles : "can work with"
+    Servers ||--|{ Scopes : "provides"
+```
+
+And the new auth flow is as follows:
+
+
+```mermaid
+sequenceDiagram
+    participant Flowauth
+    actor User
+    participant FlowAPI
+    note over User: Selects roles for token
+    User ->>+ Flowauth: roles
+    note over Flowauth: Checks user is permitted roles
+    Flowauth ->>+ User: signed token w/ roles + scopes
+    note over User: Builds query
+    User ->>+ FlowAPI: query, token
+    note over FlowAPI: Checks query params are covered by a single role
+    FlowAPI ->>+ User: query results
+```
+
+**Consequences**
+ - Much faster and more lightweight building of tokens
+ - Flowauth frontend can now be used on lower-capability machines
+ - Loss of fine-grained control over query permissions
```

Comment:
```suggestion
 - Less fine-grained control over query permissions
```

Reply:
I also think it's worth mentioning that there's a positive side to this one as well as a negative, because reducing the number of possible scopes makes decisions about what scopes should be granted to a role more manageable - the auth system consequently becomes more user-friendly (so the granularity that remains is hopefully _more_ likely to be utilised, rather than just granting all permissions because there are too many to select from).